From 6563c0b0a0b59e05d90d3819a3dff9580b78fb03 Mon Sep 17 00:00:00 2001
From: "Reddy, MulaX M" <mulax.m.reddy@intel.com>
Date: Mon, 21 Dec 2015 20:06:04 +0530
Subject: [PATCH 02/13] EM/max17042: Fix for battery SOC updation

During platform S3 sleep state, battery SOC is not
updated in user space. This creates a battery SOC
mismatch in battery sysfs and user space.

This patch is sending power_supply_changed notification
after resuming from sleep, which internally calls uevent
notification to user space. Thus user space also will be
updated appropriately.

Change-Id: I8896aa7671147655782a2640046404030316e7fa
Signed-off-by: Reddy, MulaX M <mulax.m.reddy@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-6006
Reviewed-on: https://android.intel.com:443/450991
---
 drivers/power/max17042_battery.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/power/max17042_battery.c b/drivers/power/max17042_battery.c
index dc15110..be9d7c4 100644
--- a/drivers/power/max17042_battery.c
+++ b/drivers/power/max17042_battery.c
@@ -1371,6 +1371,9 @@ static int max17042_resume(struct device *dev)
 		max17042_set_soc_threshold(chip, 1);
 	}
 
+	/* trigger the updation of battery status in user space */
+	power_supply_changed(&chip->battery);
+
 	return 0;
 }
 #endif
-- 
1.9.1

